### PR TITLE
Added improved README and new contributing file

### PR DESCRIPTION
**Pull request by langsmith**
This pr cleans up this repo's README file to match other Mapbox repos' files. It also adds a `CONTRIBUTING.md` file to the repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,16 +2,44 @@
...
- token to **flutter_mapbox/example/android/local.properties** 
-- Run app with `cd flutter_mapbox/example && flutter run`
\ No newline at end of file
+ token to the **local.properties** file.
+- Run the app with `cd flutter_mapbox/example && flutter run`
+
+
+## Documentation
+
+This README file currently houses all of the documentation for this Flutter project. Please visit [mapbox.com/android-docs](https://www.mapbox.com/android-docs/) if you'd like more information about the Mapbox Maps SDK for Android.
```
Comment (tobrun) [inline README.md]: can you add a link to iOS docs as well?

File: README.md [hunk 2]
```diff
@@ -2,16 +2,44 @@
...
+- **Have a bug to report?** [Open an issue](https://github.com/mapbox/flutter-mapbox-gl/issues/new). If possible, include a full log and information which shows the issue.
+- **Have a feature request?** [Open an issue](https://github.com/mapbox/flutter-mapbox-gl/issues/new). Tell us what the feature should do and why you want the feature.
+
+
+## Sample code
+
+This repository's own code and demo app is code for you to reference.
+
+It doesn't currently use Flutter, but you could download the [Mapbox Demo App](https://play.google.com/store/apps/details?id=com.mapbox.mapboxandroiddemo) to see what's possible with the Mapbox Maps SDK for Android. Perhaps you'll find use cases in it which you'd like to see in this Flutter project.
+
```
Comment (tobrun) [inline README.md]: Mapbox Demo App is not really relevant for this repository, maybe just add a link to [here](https://github.com/mapbox/flutter-mapbox-gl/tree/master/example/lib) for now?

Comment (langsmith) [reply README.md]: Fixed

File: README.md (hunk 1 above)
Comment (langsmith) [reply README.md]: Fixed

**Comment by langsmith** [comment]
All set @tobrun ?

**Pull request merged**